compile/service_test.go

```diff
@@ -280,6 +319,28 @@ func TestCompileServiceFailure(t *testing.T) {
 			`,
 			[]string{`function "bar" cannot`, "raise exceptions"},
 		},
+		{
+			"duplicate annotation name",
+			`
+				service AnnotatedService {
+
+				} (test = "mytest", test = "t")
+			`,
+			[]string{
+				`the name "test" has already been used`,
+			},
+		},
+		{
+			"duplicate annotation name on function",
+			`
+				service AnnotatedService {
+					i32 bar() (functest = "test" functest = "t")
```

Comment:
Although the grammar makes separators optional, you should probably add a comma between the two annotations.